### ISSUE TITLE
Project system

### ISSUE THREAD

**Issue by Raconteur32** [Brancieq-Paul/modpack-creator#1]
Add a project layer.

A project should be represented by a config file with basics data: id/name, description ect...
It should also contain configurations variable that should be used for executing steps of a workflows. The steps should prioritize the use of their configuration in the workflow.json, and use project configuration variables only if the variable is not found in the step configuration, or if the step configuration refer to the project configuration.

Example:

Project configuration:
{
    "value_referenced": "This value will be referenced in step configuration"
    "overrided_value_configuration": "This configuration value will be overrided in the step config"
    "non_overrided_value": "This config value will not be overrided by the step configuration"
}

Step configuration (will be used in priority):
{
    "step_specific_configuration": "Value only needed for this execution of this step",
    "configuration_from_project_reference": "ref:value_referenced",
    "overrided_value_configuration": "Value from here override project configuration value"
}

To summarize, the step at the end will be able to access values of value_referenced, non_overrided_value from the project config, and step_specific_configuration, overrided_value_configuration from the step config.

Taks:
- [x] Creating a class to handle stack of config and references
- [x] Creating a project.json file and a Project class (with user config and project variables), compile into project config MultiLayer object
- [x] Add the multilayer logic into workflows and steps
- [x] Test logic with test project and test workflow
